cli/internal/taskhash/taskhash.go

```diff
@@ -62,44 +62,13 @@ func NewTracker(rootNode string, globalHash string, pipeline fs.Pipeline) *Track
 	}
 }
 
-// packageFileSpec defines a combination of a package and optional set of input globs
-type packageFileSpec struct {
-	pkg    string
-	inputs []string
+// packageFileHashInputs defines a combination of a package and optional set of input globs
+type packageFileHashInputs struct {
+	taskID         string
+	taskDefinition *fs.TaskDefinition
+	packageName    string
 }
 
-func specFromPackageTask(packageTask *nodes.PackageTask) packageFileSpec {
-	return packageFileSpec{
-		pkg:    packageTask.PackageName,
-		inputs: packageTask.TaskDefinition.Inputs,
-	}
-}
-
-// packageFileHashKey is a hashable representation of a packageFileSpec.
-type packageFileHashKey string
-
-// hashes the inputs for a packageTask
-func (pfs packageFileSpec) ToKey() packageFileHashKey {
-	sort.Strings(pfs.inputs)
-	return packageFileHashKey(fmt.Sprintf("%v#%v", pfs.pkg, strings.Join(pfs.inputs, "!")))
-}
-
-func (pfs *packageFileSpec) getHashObject(pkg *fs.PackageJSON, repoRoot turbopath.AbsoluteSystemPath) (map[turbopath.AnchoredUnixPath]string, error) {
-	return hashing.GetPackageFileHashes(repoRoot, pkg.Dir, pfs.inputs)
-}
-
-func (pfs *packageFileSpec) hash(hashObject map[turbopath.AnchoredUnixPath]string) (string, error) {
-	hashOfFiles, otherErr := fs.HashObject(hashObject)
```

Comment:
`fs.HashObject` is actually infallible, in spite of the signature.

From `xxhash.Write`:

> `Write` adds more data to `d`. It always returns `len(b), nil`.